### ISSUE TITLE
Git does not appear to be working

### ISSUE THREAD

**Issue by liam-potter** [Microsoft/vscode#38427]
- VSCode Version: Code - Insiders 1.19.0-insider (73830a36872aa1c168c085dad8cc6e2e1a05ff45, 2017-11-13T15:41:17.934Z)
- OS Version: Windows_NT x64 10.0.15063

Steps to Reproduce:
1. Open any project in which Git has been initialised (i.e. project has a valid `.git` folder)
2. Open Source Control panel (Ctrl + Shift + G) and observe note "There are no active source control providers."
3. Observe `git status` in integrated terminal works fine (in my case I am using integrated bash)
4. Observe issue persists across multiple projects, so issue is not specific to a corrupted individual `.git` folder.

Reproduces without extensions: Yes

**Comment by liam-potter** [comment]
Just upgraded to the following version, the issue does not appear to be resolved:

Code - Insiders 1.19.0-insider (3a7d517f65212de615f469587c2ace53275ce96e, 2017-11-15T05:17:28.098Z)

**Comment by wangdachangchang** [comment]
@liam-potter I upgraded this version just now, but git does not appear,
VSCode Version : Code - Insiders 1.19.0-insider (3a7d517, 2017-11-15T05:17:28.098Z)
what should i do for this?

**Comment by liam-potter** [comment]
@wangdachangchang My comment was that upgrading to that version **did not** resolve the issue.

I have upgraded to the following version this morning and Git appears to be working again:

(91a2020698b5220e50f60682724927a810608356, 2017-11-16T05:17:48.036Z)

I will test this on my work machine (where Git was broken) tomorrow to confirm it is definitely fixed.

**Comment by joaomoreno** [comment]
https://github.com/Microsoft/vscode/issues/38244#issuecomment-344027257

Please use `git.path` to configure where `git` is located.

**Comment by liam-potter** [comment]
I have never had Github for Windows installed on my work machine, only Git, and after arriving into the office this morning this issue persisted.

Reinstalling Git and updating `git.path` manually in my settings file was the only fix. Under no circumstances could I get it to automatically pull Git from PATH.

**Comment by joaomoreno** [comment]
@liam-potter Got it. Can you do me a favor? Search in the registry for the path where git.exe is installed. Maybe if we can look into that registry value, we could detect where you have installed it.

**Comment by liam-potter** [comment]
Sorry if I wasn't being clear, but the issue is resolved for me now that I have manually configured my `git.path` setting.

As requested I have looked in the registry but the only references to `git.exe` I could find were session-based values under `Computer\HKEY_USERS\S-1-5-21-1119021569-3052555868-3407816479-1157\Software\Microsoft\RestartManager\Session0001`. It has a `RegFiles0000` key of `REG_MULTI_SZ` type and a list of various git-based `dll`s/`exe`s. Not sure if that would be helpful?
